### PR TITLE
DFPL-1975 Rewrite adminAddCaseFlag_test.js in Playwright

### PR DESCRIPTION
**Pull request by GauravChawlaMOJ**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-1975
https://tools.hmcts.net/jira/browse/DFPL-2278

### Change description ###

admin add/remove case flag test added

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: playwright-e2e/pages/add-and-remove-admin-case-flag.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        this.yes =  page.getByLabel('Yes');
+        this.uploadAssessmentForm = page.getByRole('textbox', { name: 'Upload assessment form or' });
+        this.additionalNotes =  page.getByLabel('Additional notes (Optional)');
+        this.addOrRemoveFlag =  page.getByRole('link', { name: 'Add or remove case flag' });
+        this.no =  page.getByLabel('No', { exact: true });
+    }
+
+    public async addCaseFlag() {
+        await this.yes.check();
+        await this.uploadAssessmentForm.setInputFiles(config.testPdfFile);
```
Comment (DanCatchpole) [inline playwright-e2e/pages/add-and-remove-admin-case-flag.ts:24]: do we need to have the wait for upload to be finished code after this?

File: playwright-e2e/settings/user-credentials.ts [hunk 2]
```diff
@@ -28,6 +28,10 @@ export const judgeUser = {
   email: process.env.JUDGE_USERNAME || 'judiciary-only@mailnesia.com',
   password: defaultPwd,
 };
+export const judgeUserWithAdminRole = {
+    email: process.env.JUDGE_USER_WITH_ADMIN || 'judiciary@mailnesia.com',
+    password: defaultPwd,
+};
```
Comment (DanCatchpole) [inline playwright-e2e/settings/user-credentials.ts:34]: I don't think we should be using this judge anymore, legacy judges will not have the correct roles moving forwards (due to interactions with AM + JRD), should be a CTSC user if admin actions like this event I think, or judge wales (or equiv) for judicial actions

File: playwright-e2e/tests/admin-add-case-flag.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            await signInPage.login(CTSCUser.email, CTSCUser.password);
+            await runTest(signInPage, addAdminCaseFlag, page);
+        });
+
+    test('Add and remove a case flag as judicial admin user',
+        async ({page, signInPage, addAdminCaseFlag}) => {
+            caseName = 'Add and remove a case flag' + dateTime.slice(0, 10);
+            await apiDataSetup.updateCase(caseName, caseNumber, caseData);
+            await signInPage.visit();
+            await signInPage.login(judgeUserWithAdminRole.email, judgeUserWithAdminRole.password);
```
Comment (DanCatchpole) [inline playwright-e2e/tests/admin-add-case-flag.spec.ts:33]: can use normal judgeWales here as mentioned in the other comment!

File: playwright-e2e/settings/user-credentials.ts (hunk 2 above)
Comment (GauravChawlaMOJ) [reply playwright-e2e/settings/user-credentials.ts:34]: cool thanks, removed this user.

File: playwright-e2e/tests/admin-add-case-flag.spec.ts (hunk 3 above)
Comment (GauravChawlaMOJ) [reply playwright-e2e/tests/admin-add-case-flag.spec.ts:33]: done

File: playwright-e2e/pages/add-and-remove-admin-case-flag.ts (hunk 1 above)
Comment (GauravChawlaMOJ) [reply playwright-e2e/pages/add-and-remove-admin-case-flag.ts:24]: added wait

**Review by DanCatchpole — APPROVED** [review]
lgtm!

**Pull request merged**